programs/storage_api/src/storage_contract.rs

```diff
@@ -45,3 +54,303 @@ pub enum StorageContract {
         reward_validations: Vec<Vec<CheckedProof>>,
     },
 }
+
+pub trait StorageAccount {
+    fn submit_mining_proof(
+        &mut self,
+        sha_state: Hash,
+        entry_height: u64,
+        signature: Signature,
+    ) -> Result<(), InstructionError>;
+    fn advertise_storage_recent_blockhash(
+        &mut self,
+        hash: Hash,
+        entry_height: u64,
+    ) -> Result<(), InstructionError>;
+    fn proof_validation(
+        &mut self,
+        entry_height: u64,
+        proofs: Vec<CheckedProof>,
+        replicator_accounts: &mut [KeyedAccount],
+    ) -> Result<(), InstructionError>;
+    fn claim_storage_reward(
+        &mut self,
+        entry_height: u64,
+        tick_height: u64,
+    ) -> Result<(), InstructionError>;
+}
+
+pub trait State<T> {
```

Comment:
Why copy-paste this code?

Reply:
Oh yeah I was going to ask you where to move this. In the instruction util file?

Reply:
Yep, definitely. If you can think of a name with more precision, that'd be cool.  Pretty much anything could be called State. I regret adding that suffix to all the instruction processor state.  Not sure if Contract is the right term.